### DISCUSSION TITLE
Listening to a Livewire event inside x-init

### DISCUSSION THREAD

**Discussion by barrydevt** [livewire/livewire#5232] [1. Help]
In my Livewire component I have:

```
    public function selectGroup($value)
    {
        $this->dispatchBrowserEvent('refreshlive');
    }
```

In my Alpine component, I can detect this event successfully like so:

```
<div
     wire:ignore
     x-data="{
       options: @entangle($options),
       value: @entangle($model),
       debounce: null,
     }"
    @refreshlive.window="alert('Heard you!')"
></div>
```

This fires the `Heard you!` alert whenever the PHP method is fired. Cool.

I would like to declare the event listener in my x-init (so it can interact with Choicesjs which I am trying to get to play nice.) I have tried amongst other things:

```
<div
     wire:ignore
     x-data="{
       options: @entangle($options),
       value: @entangle($model),
       debounce: null,
     }"

    x-init="
      $nextTick(() => {
        $wire.on('refreshlive', () => {
          alert('Can I hear you?');
        })
    }"
>
</div>
```
But the alert in this case never fires. 

Any pointers?

**Comment by jonneroelofs** [comment]
I think the problem is that in your last code snippet you are actually listening for a livewire event and not a browser event. So it would probably work if you change dispatchBrowserEvent() to emit() in your livewire component. 

As you are already entangling your model you probably don't need to fire an event, in Alpine you can create a watcher that checks if the value changes and then update choicesjs accordingly and vice versa update the value property of your alpine component when a user selects an options with choicejs.
